jsonrpc/filter_manager.go

```diff
@@ -342,6 +349,116 @@ func (f *FilterManager) Exists(id string) bool {
 	return ok
 }
 
+// GetLogsForQuery return array of logs for given query
+func (f *FilterManager) GetLogsForQuery(query *LogQuery) ([]*Log, error) {
+	result := make([]*Log, 0)
+	parseReceipts := func(block *types.Block) error {
+		receipts, err := f.store.GetReceiptsByHash(block.Header.Hash)
+		if err != nil {
+			return err
+		}
+
+		for indx, receipt := range receipts {
+			for logIndx, log := range receipt.Logs {
+				if query.Match(log) {
+					result = append(result, &Log{
+						Address:     log.Address,
+						Topics:      log.Topics,
+						Data:        argBytes(log.Data),
+						BlockNumber: argUint64(block.Header.Number),
+						BlockHash:   block.Header.Hash,
+						TxHash:      block.Transactions[indx].Hash,
+						TxIndex:     argUint64(indx),
+						LogIndex:    argUint64(logIndx),
+					})
+				}
+			}
+		}
+
+		return nil
+	}
+
+	if query.BlockHash != nil {
+		block, ok := f.store.GetBlockByHash(*query.BlockHash, true)
+		if !ok {
+			return nil, fmt.Errorf("not found")
+		}
+
+		if len(block.Transactions) == 0 {
+			// no txs in block, return empty response
+			return result, nil
+		}
+
+		if err := parseReceipts(block); err != nil {
+			return nil, err
+		}
+
+		return result, nil
+	}
```

Comment:
This can be extracted out (since it's practically the same logic you do in a range)

cc @dbrajovic

Reply:
Yes, but because the one uses GetBlockByHash and the other GetBlockByNumber I would need to pass multiple params and have condition inside it. Can discuss tmrw about it

Reply:
Also most of these test in the jsonrpc plus the whole filter_menager should be refactored. So i would not add more utility methods if we are going to refactor it in general.